platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java

```diff
@@ -1211,4 +1260,46 @@ public <T extends SwirldState> AutoCloseableWrapper<T> getLatestSignedState(@Non
     private boolean isLastEventBeforeRestart(final EventImpl event) {
         return event.isLastInRoundReceived() && swirldStateManager.isInFreezePeriod(event.getConsensusTimestamp());
     }
+
+    /**
+     * Inform all components that a fatal error has occurred, log the error, and shutdown the JVM.
+     */
+    private void handleFatalError(
+            @Nullable final String msg, @Nullable final Throwable throwable, @NonNull final SystemExitCode exitCode) {
```

Comment:
Could we define this method in a static helper class somewhere? Just looking for ways to reduce the amount of business logic that lives inside this file.

Reply:
this is something that unfortunately needs to be in platform. ATM, it needs to have a reference to the state management component to tell it to dump its state. but in the future, it might need to do other things. I have simplified it a lot though, so its pretty small now